api/types/src/wrappers.rs

```diff
@@ -104,5 +110,80 @@ impl fmt::Display for MoveStructTagWrapper {
     }
 }
 
-impl_poem_type!(MoveStructTagWrapper);
 impl_poem_parameter!(MoveStructTagWrapper);
+
+// Currently it is not possible to deserialize certain MoveTypes, such as
+// generic type params. In those cases, we give up on parsing them as
```

Comment:
For starters, the underlying function within the move create just doesn't handle generic type params at all. For this to work properly, I'd need to be able to inject additional parsing capabilities into the function so when it eventually parses down to the generic type param token, it knows what to do with it. Even this isn't enough because the enum they use doesn't have this as an option. So if we wanted to support it we'd have to add rewrite a lot of stuff ourselves (unless we got this support into move directly, but Victor says it doesn't really make sense there because it's only meant for concrete types).